Sources/Paywalls/Components/PaywallStickyFooterComponent.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallStickyFooterComponent.swift
+//
+//  Created by Jay Shortway on 24/10/2024.
+//
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    struct StickyFooterComponent: PaywallComponentBase {
```

Comment:
I'm now starting to think whether we actually need this, or if we can get away with just using the existing `StackComponent`. Are there any properties that a sticky footer needs that a stack doesn't? Maybe a shadow? But one could argue that a regular stack can also have a shadow.

Reply:
Oh! Maybe this can be a `StickyComponent` with and `edge` property? That way we can use it for a sticky header as well (and sticky sidepanel if we _ever_ want that).

Reply:
Ohhhh... this is a great question!

I think it might actually be almost entirely a stack but... It could be beneficial to keep as its own object if we do need to ever add anything specific (like translucent or a drawer type of functionality or something 🤷‍♂️

Thoughts?

Reply:
Yea it's definitely safer to keep it separate like this, to be able to add any properties we might think of in the future. The JSON will get more nested, but I'm not super worried about that haha. Dunno 'bout you?